app/presenters/publishing_api/historical_account_presenter.rb

```diff
@@ -41,6 +41,42 @@ def links
         person: [
           historical_account.person.content_id,
         ],
+        ordered_related_items: related_pms,
+      }
+    end
+
+    def related_pms
+      role = Role.friendly.find("prime-minister")
+      appointments = RoleAppointment.where(role_id: role.id).distinct(&:person).reorder("started_at DESC")
+      people = appointments.map(&:person_id)
+      person_id = historical_account.person.id
+
+      related_person_id_finder(people, person_id).map do |id|
+        person_hash_builder(id)
+      end
+    end
+
+    def related_person_id_finder(people_ids, person_id)
+      person_index = people_ids.index(person_id)
+      starting_index = person_index - 3
+      ending_index = person_index + 2
+
+      people = if starting_index.negative?
+                 people_ids.first(6)
+               elsif ending_index >= people_ids.length
+                 people_ids.last(6)
+               else
+                 people_ids[starting_index..ending_index]
+               end
+      people - [person_id]
+    end
+
+    def person_hash_builder(id)
```

Comment:
love the first bit, have implemented. Going to leave the 2nd bit because it would mean loading up all the people who've ever held the role rather than just looking up the closest few. Granted only tiny computational saving so may not be worth it

Reply:
Sure, sounds sensible to me!

Reply:
It does feel like there's a lot of swapping back and forth between IDs and objects here, when we could just use the objects throughout.

I had a quick go at refactoring to just use objects and came up with this (based on the original code on this PR, not the changes you've subsequently made):
```ruby
    def related_pms
      all_appointees = Role
        .friendly
        .find("prime-minister")
        .role_appointments
        .order(:started_at)
        .map(&:person)
        .uniq
      person_to_present = historical_account.person

      related_people(all_appointees, person_to_present).map do |person|
        {
          "title" => person.name,
          "base_path" => person.historical_accounts.present? ? person.historical_accounts.first.public_path : "/government/history/prime-ministers",
        }
      end
    end

    def related_people(all_appointees, person)
      person_index = all_appointees.index(person)
      starting_index = person_index - 3
      ending_index = person_index + 2

      people = if starting_index.negative?
                 all_appointees.first(6)
               elsif ending_index >= all_appointees.length
                 all_appointees.last(6)
               else
                 all_appointees[starting_index..ending_index]
               end
      people - [person]
    end
```

Reply:
yeah, i think it's the swapping that had me questioning the code in the first place. I like the refactor but it's going to take me a little while to fix the tests

Reply:
scratch that, tests were fine. Just waiting for the tests to run